README.md
# Superdesk Analytics
_Jan 2017_

[![Build Status](https://travis-ci.org/superdesk/superdesk-analytics.svg?branch=master)](https://travis-ci.org/superdesk/superdesk-analytics)
[![Coverage Status](https://coveralls.io/repos/github/superdesk/superdesk-analytics/badge.svg?branch=master)](https://coveralls.io/github/superdesk/superdesk-analytics?branch=master)

## Overview
This is a plugin for [superdesk](https://github.com/superdesk/superdesk).

It adds an analytics application page with its API endpoints.

## License Notice
You must have a valid license for [Highcharts](https://www.highcharts.com/) JS v6.x to use this plugin.

The Highcharts JS library is used to render the charts. It is available under different licenses depending on whether it is intended for commercial/government use, or for a personal or non-profit project.
[see more](https://shop.highsoft.com/faq#Non-Commercial-0)


## Table of contents
* [Installation](#installation)
    * [Client](#client-enable-the-superdesk-analytics-module)
    * [Server](#server-load-the-superdesk-analytics-module)
    * [Development](#development-setup)
* [Highcharts License](#highcharts-license)
* [Config Options](#config-options)
* [Highcharts Export Server](#highcharts-export-server)
    * [Installing the Service](#installing-the-service)
    * [Running the Service](#running-the-service)
    * [Configure Process Workers](#configuring-process-workers)
    * [Rate Limiting](#rate-limiting--highcharts-requests)
* [Scheduled Reports](#scheduled-reports)
* [Archive Statistics](#archive-statistics)
* [Reports](#archive-reports)
    * [Archive Reports](#archive-reports)
        * [Content Publishing](#content-publishing)
        * [Publishing Performance](#publishing-performance)
    * [Advanced Archive Reports](#advanced-archive-reports)<sup>1</sup>
        * [Desk Activity](#desk-activity)
        * [Featuremeida Updates](#featuremedia-updates)
        * [Production Time](#production-time)
        * [Update Time](#update-time)
        * [User Activity](#user-activity)
    * [Planning Module Reports](#planning-module-reports)
        * [Planning Usage](#planning-usage)

1: These reports require [archive statistics](#archive-statistics) stats to be enabled

## Installation

In order to see the analytics feature in the application, you need to enable it.

### Client: Enable the superdesk-analytics module
In `client/superdesk.config.js` from `superdesk`, add this line
```js
features : {analytics: true},
```
This will import the `superdesk-analytics` node module and load the `superdesk.analytics` angular module in the main angular application.

Add the following line in the package.json file, dependencies:
```
"superdesk-analytics": "superdesk/superdesk-analytics#1.5"
```

Run the command:
```
npm install
```

This will install the analytics module.

### Server: Load the superdesk-analytics module
```diff
--- a/server/settings.py
+++ b/server/settings.py
@@ -132,7 +131,8 @@ INSTALLED_APPS.extend([
+    'analytics',
 ])
```

Add the following line in the requirements.txt file:
```
git+git://github.com/superdesk/superdesk-analytics.git@1.5#egg=superdesk-analytics
```

Run the command:
```
pip install -r requirements.txt
```

This will install the analytics package.


### Development setup

Download and install
```
git clone git@github.com:superdesk/superdesk-analytics.git
make install
```
Run the tests
```
source server/env/bin/activate
make test
```

Connect the repository to `superdesk-client-core`
```
npm link
cd /path/to/superdesk-client-core
npm link superdesk-analytics
```

Connect the repository to `superdesk-core`
```
cd /path/to/superdesk-core
source env/bin/activate
pip install -e /path/to/superdesk-analytics
```


## Highcharts License
You must have a valid license for [Highcharts](https://www.highcharts.com/) JS v6.x to use this plugin.

The Highcharts JS library is used to render the charts. It is available under different licenses depending on whether it is intended for commercial/government use, or for a personal or non-profit project.

To have the license details available to the end user in the Analytics page of Superdesk, provide the following config options in your settings.py:

* HIGHCHARTS_LICENSE_TYPE (High-Five, Develop or OEM)
* HIGHCHARTS_LICENSEE (Name of the entity that owns the license)
* HIGHCHARTS_LICENSEE_CONTACT (A contact email address for the licensee)
* HIGHCHARTS_LICENSE_ID (The license ID provided by Highsoft)
* HIGHCHARTS_LICENSE_CUSTOMER_ID (A custom license field to use an internal customer number, if required)
* HIGHCHARTS_LICENSE_EXPIRY (the expiry of the license)


## Config Options
* HIGHCHARTS_SERVER_HOST (defaults to 'localhost')
* HIGHCHARTS_SERVER_PORT (defaults to '6060')
* HIGHCHARTS_SERVER_WORKERS (defaults to 4) - Number of workers to spawn
* HIGHCHARTS_SERVER_WORK_LIMIT (defaults to 60) - The pieces of work that can be performed before restarting a phantom process
* HIGHCHARTS_SERVER_LOG_LEVEL (defaults to 3) - Set the log level. Available options are:
* HIGHCHARTS_SERVER_QUEUE_SIZE (defaults to 10) - how many request can be stored in overflow count when there are not enough
* HIGHCHARTS_SERVER_RATE_LIMIT (defaults to False) - The max requests allowed in one minute
* ANALYTICS_ENABLE_SCHEDULED_REPORTS (defaults to False) - Enable the emailing of scheduled reports
* ANALYTICS_ENABLE_ARCHIVE_STATS (defaults to False)
* STATISTICS_MONGO_DBNAME (defaults to 'statistics')
* STATISTICS_MONGO_URI (defaults to 'mongodb://localhost/statistics')
* STATISTICS_ELASTIC_URL (defaults to ELASTICSEARCH_URL config)

## Highcharts Export Server
To be able to generate charts on the server, we need to install/run the Highcharts Export Server.

### Installing the service
```
npm install -g highcharts-export-server
```

There is also a script that can be used to automate the install:
```
cd server/scripts && ./install-highcharts-export-server.sh
```
This will automatically accept the end user licence, and use a specific version of highcharts
* ACCEPT_HIGHCHARTS_LICENSE=1
* HIGHCHARTS_VERSION=6.2.0
* HIGHCHARTS_USE_STYLED=1
* HIGHCHARTS_MOMENT=1

### Running the service
There is a python module to allow running the highcharts export server.
```
python3 -u -m analytics.reports.highcharts_server
```

This will start the service using the host/port configured in settings.py (relative to your current working directory)
* HIGHCHARTS_SERVER_HOST (defaults to 'localhost')
* HIGHCHARTS_SERVER_PORT (defaults to '6060')

### Adding the service to a Honcho Procfile
You can add the service to your Honcho Procfile with the following line:
```
highcharts: python3 -u -m analytics.reports.highcharts_server
```

### Configuring Process Workers
The export server uses a pool of PhantomJs worker threads. You can configure this pool in settings.py
* HIGHCHARTS_SERVER_WORKERS (defaults to 4) - Number of workers to spawn
* HIGHCHARTS_SERVER_WORK_LIMIT (defaults to 60) - The pieces of work that can be performed before restarting a phantom process
* HIGHCHARTS_SERVER_LOG_LEVEL (defaults to 3) - Set the log level. Available options are:
    * 0 - off
    * 1 - errors
    * 2 - warn
    * 3 - notice
    * 4 - verbose
* HIGHCHARTS_SERVER_QUEUE_SIZE (defaults to 10) - how many request can be stored in overflow count when there are not enough

### Rate limiting  Highcharts requests
The highcharts export server has the ability to rate limit the requests that it receives.
By default rate limiting is turned off. You can turn it on in your settings.py
* HIGHCHARTS_SERVER_RATE_LIMIT (defaults to False) - The max requests allowed in one minute


## Scheduled Reports
To enable reports to be periodically scheduled (emailed), you must enable the config in settings.py.
If this is enabled, then the celery queue entry will be created.
* ANALYTICS_ENABLE_SCHEDULED_REPORTS (defaults to False) - Enable the emailing of scheduled reports

## Archive Statistics
Archive statistics are generated from the `archive_history` collection and stored in an `archive_statistics` collection. This allows in depth reports for content, desk and user activities.

Generating these stats are disabled by default (due to resource requirements), and can be enabled in settings.py.:
* ANALYTICS_ENABLE_ARCHIVE_STATS=True

The data for archive statistics is stored in a separate mongodb database and elastic index. These can be configured with the following:
* STATISTICS_MONGO_DBNAME (defaults to 'statistics')
* STATISTICS_MONGO_URI (defaults to 'mongodb://localhost/statistics')
* STATISTICS_ELASTIC_URL (defaults to ELASTICSEARCH_URL config)

Without enabling the archive stats, the following reports will be disabled:
* [Desk Activity](#desk-activity)
* [Featuremeida Updates](#featuremedia-updates)
* [Production Time](#production-time)
* [Update Time](#update-time)
* [User Activity](#user-activity)

Statistics are generated using the following celery beat schedule:
* CELERY_BEAT_SCHEDULE\['analytics:gen_archive_stats'\]

If the above is not defined, then it will default to run at 3am every day


## Archive Reports

### Content Publishing
As an administrator I would like to be able to generate metadata reports around content publishing.

I would like to be able to group, and optionally subgroup, the report by the following metadata:
* Source
* Category
* News Value
* Genre

The report should have the ability to filter data based on the following:
* Date (yesterday, last week, last month, range)
* Desk
* User
* Published State (published, killed, corrected, recalled, rewrite/update)

I should be able to choose one of the following chart types for the report:
* Bar
* Column
* Table

### Publishing Performance
As an administrator I would like to generate a report detailing Publishing Performance.

I would like to be able to group the report by one the following:
* Desk
* User
* Category
* Source
* News Value
* Genre

Each group should be broken down to the following publish states:
* Kills
* Corrections
* Rewrites
* Updates
* Originals

## Advanced Archive Reports

### Desk Activity
As an administrator I would like to be able to analyse peak times on a desk.

This report will show the total incoming and outgoing actions on the desk, grouped by hour or by day (based on a drop down field on the form). These actions should include:

Incoming:
* Manually created by a user
* Sent to the desk (by the send to action)
* Fetched from ingested content
* Unspiked
* Duplicated
* System routed

Outgoing:
* Published
* Sent from the desk (by the send to action)
* Spiked

Below the chart should be a count of all the actions that made up this report, displayed in a table.

### Featuremedia Updates
As an administrator I would like to be able to analyse updates to story attachments.

This report will display a table of each update to stories that have asset's attached to them.

The updates will include:
* Pictures added
* Pictured removed
* Pictured altered

Links should be provided to the images and stories so the user can view the image in detail.

### Production Time
As an administrator I would like to be able to analyse time spent on stories..

This report will provide statistics on the minimum, average and maximum amount of time spent on producing stories for desks.

These figures will be calculated based on when a story enters a desk, and when that same story exits the desk.

Incoming:
* Manually created by a user
* Sent to the desk (by the send to action)
* Fetched from ingested content
* Unspiked
* Duplicated
* System routed

Outgoing:
* Published
* Sent from the desk (by the send to action)
* Spiked

### Update Time
As an administrator I would like to be able to analyse updates to 3 par stories.

I expect this report to show how long it took to create these updates.

### User Activity
This report will show a breakdown of a users workload during a single day, broken down by the individual stories that they worked on.

I expect to be able to click on any single story in the chart, that will then display a second chart displaying the life-cycle of that story, from initial creation to the last action taken on the story

## Planning Module Reports

### Planning Usage
As an administrator I would like to be able to analyse who is using Planning.

I expect this report to include the list of users who are using Planning, and a list of users who are not using Planning.

The report will include counts per user for the following actions:
* Planning items created
* Coverages created
* Coverages that are assigned to workflow? - (do we need this)
* Events created? - (do we need this)

I also expect that users who do not have permission to use Planning not to be included in this report.






CHANGELOG.md
# Superdesk Analytics Changelog

## [v2.6.0-rc1] 2022-12-12
### Improvements
- chore: Enable CI on Python 3.10 (#143)

### Fixes
- fix(build): Replace sass table border var with css var (8fda4d0)
- fix(flake8): Remove hash from config (8bd3b7b)
- fix(mypy): Add optional to _run_highcharts_cli signature (166e68b)

## [2.5.1] 2022-10-28
### Fixes
- [SDESK-6588] fix: Make table reports use full width available (#144)

## [2.5.0] 2022-10-28
### Fixes
- [SDESK-6489] fix: Copy ContentAPI es settings for Statistics (#140)
- [SDESK-6490] fix: Use Highcharts cli instead of the HTTP server (#141)
- [SDESK-6490] fix: Install highcharts cli locally (#142)

## [2.1.2] 2021-03-03
### Fixes
- fix(schema): Production Time not returning data (#137)

## [2.1.1] 2021-02-26
### Fixes
- [SDESK-5839] fix(elastic): Show more than 10 entries in reports (#136)

## * Release 2.1.0 - 2021-02-11

## [2.1.0-rc.1] 2021-01-28
- Update Superdesk & Planning to release/2.1 branch

## [2.0.0] 2020-10-28
### Features

### Improvements
- elastic7 support (#129)

### Fixes
- [SDESK-5471] Minor UI issue when creating analytics report schedule. (#134)
- [SDESK-5205] fix errors after eve/flask upgrade (#124)

## [1.33.5] 2020-10-01
- Bump ui-framework version

## [1.33.4] 2020-09-17
- Release 1.33.4

## [1.33.4-rc1] 2020-09-16
### Fixes
- move jquery and gridster to dev dependencies

## [1.33.3] 2020-08-20
- Release 1.33.3

### Bugs
- Incorrect versions published to npmjs and pypi

## [1.33.2] 2020-08-19
- Release 1.33.2

## [1.33.2-rc1] 2020-08-18
### Improvements
- [SDESK-5322] Change Recall labels to Takedown (#130)
- [SDCP-271] Display desks with zero new stories in Superdesk Analytics reports (#132)
- [SDCP-270] Add ContentType filter to PublishingPerformance report. (#133)

### Fixes
- [SDESK-5392] fix: Incorrect moment import in utils (#131)

## [1.33.1] 2020-06-10
- Release 1.33.1

## [1.33.1-rc1] 2020-06-04
### Features
- [SDESK-5220] Modal to show Highcharts license (#127)

### Improvements
- [SDESK-3882] Replace ng-gettext with gettext from superdesk (#125)
- [SDESK-5276] Add group by author to Publishing Performance report (#128)

### Bugs
- [SDESK-5234] Hide Analytics menu if no reports available (#126)

## [1.33.0] 2020-04-28
- Release 1.33.0

## [1.33.0-rc1] 2020-04-22
### Fixes
- [SDESK-5133] Use appConfig instead of config/deployConfig (#121)
- [SDESK-5146] Clear filter values upon clicking the button (#122)
- fix(appConfig): Improper path used when importing appConfig (#123)

## [1.7.4] 2020-03-03
### Fixes
- fix(build): Use github/commit for highcharts export server version (#118)

## [1.7.3] 2020-01-31
### Fixes
- [SDESK-4847] Implement CSV download for tables (#113)
- [SDESK-4695] Convert relative dates to absolute for date_histogram bounds (#112)
- [SDESK-4695] (fix): Histogram aggregations failing for relative dates (#114)
- fix(install): Fix installing mkdir for highcharts-export-server (#115)
- fix(packages): Update versions based on GitHub recommendations (#116)

## [1.7.2] 2019-11-05
### Improvements
- [SDESK-4779] Add subject to publishing based reports (#111)

### Bug Fixes
- Set cores to Superdesk v1.30 (#108)
- (fix): rewritten_by was not being cleared in stats (#110)


## [1.7.1] 2019-09-16
### Bug Fixes
- (fix) flake tests (#107)
- [SDESK-4289](fix): Missing fields stored in Archive Statistics (#106)
- [SDESK-4277] Include all results for the Planning Usage Report (#105)
- fix signals import (#104)
- [SDESK-3998] Include updates for featuremedia stats (#103)


## [1.7.0] 2019-05-06
### Features
- [SDESK-3981] Provide signals for generating custom stats
- [SDESK-3947] Make user filters searchable
- Remove inactive/disabled users from the dropdown list
- [SDESK-4144] Change translations for item state
- Rearrange table heading order for PublishingPerformance report
- Add _id config schema
- [SDESK-4148] Ability to manage available date filters/chart types
- [SDESK-3799] Document 'gen_archive_statistics' command

### Bug Fixes
- (fix): Retrieve proper elastic index for statistics
- [SDESK-3947] (fix): Disabled users appearing in filters
- [SDESK-4197] (fix): Incorrect directive name for sd-report-preview-proxy
- (fix) Fixing the date histogram extended bounds parameters.
- [SDESK-4196] (fix): Cannot view schedules using the Schedule button
- [SDESK-4148] (fix): Don't import gettext from client-core
- gettext from client-core utilities is not available in 1.28
- [SDESK-3960] (fix): Etag error on patching Saved Reports
- [SDESK-4140] (fix): Don't allow saving or scheduling reports if not privileged
- [SDESK-3954] (fix): Incorrect date format in DeskActivity aggregation results
- [SDESK-4127] (fix): Export menu hidden behind table headers
- [SDESK-3938][SDESK-3958][SDESK-3957] (fix): Incorrect repo and size values
- Fix stats for removing of featuremedia


## [1.6.0] 2019-02-08
### Features
- SDESK-3799 - Cleanup and document commands in manage.py
- Generate archive stats hourly by default
- SDESK-3866 - Removed legacy reports
- SDESK-3865 - Removed legacy widgets
- SDESK-3935 - New Publishing Actions dashboard widget
- SDESK-3936 - Enhance user experience
  - Removed 'OrderBy' option for timeline based reports
  - Remove point labels (prefer tooltips instead)
  - Fix UI issue with page numbers in table reports
  - Added loading indicator while loading report and generating the chart
  - Disabled SVG/CSV in scheduled reports (currently broken)
  - Placed hour labels on top and bottom of the x-axis in the UserActivity report
  - Reduced whitespace in the chart in the UserActivity Report
  - Changed default chart types to Column or Table depending on the chart type
  - Enabled more chart types for some reports (line, area, scatter)


## [1.5.0] 2019-01-21
### Features
- SDESK-3690 - ProductionTime Report
- SDESK-3691 - User Acitivity Report
- SDESK-3692 - Featuremedia Update Report
- SDESK-3535 - Update Time Report
- (enhancement) Use superdesk-code-style for eslint config


## [1.4.0] 2019-01-03

server/settings.py
#!/usr/bin/env python
# -*- coding: utf-8; -*-
#
# This file is part of Superdesk.
#
# Copyright 2013, 2014, 2015 Sourcefabric z.u. and contributors.
#
# For the full copyright and license information, please see the
# AUTHORS and LICENSE files distributed with this source code, or
# at https://www.sourcefabric.org/superdesk/license


import os
import json
from urllib.parse import urlparse
from superdesk.default_settings import strtobool


def env(variable, fallback_value=None):
    env_value = os.environ.get(variable, "")
    if len(env_value) == 0:
        return fallback_value
    else:
        if env_value == "__EMPTY__":
            return ""
        else:
            return env_value


ABS_PATH = os.path.abspath(os.path.dirname(__file__))
INIT_DATA_PATH = os.path.join(ABS_PATH, "data")

LOG_CONFIG_FILE = env("LOG_CONFIG_FILE", "logging_config.yml")

APPLICATION_NAME = env("APP_NAME", "Superdesk")
server_url = urlparse(env("SUPERDESK_URL", "http://localhost:5000/api"))
CLIENT_URL = env("SUPERDESK_CLIENT_URL", "http://localhost:9000")
URL_PROTOCOL = server_url.scheme or None
SERVER_NAME = server_url.netloc or None
URL_PREFIX = server_url.path.lstrip("/") or ""
if SERVER_NAME and SERVER_NAME.endswith(":80"):
    SERVER_NAME = SERVER_NAME[:-3]

INSTALLED_APPS = [
    "apps.auth",
    "superdesk.roles",
]

# LDAP settings
LDAP_SERVER = env("LDAP_SERVER", "")  # Ex: ldap://sourcefabric.org
LDAP_SERVER_PORT = env("LDAP_SERVER_PORT", 389)

# Fully Qualified Domain Name. Ex: sourcefabric.org
LDAP_FQDN = env("LDAP_FQDN", "")

# LDAP_BASE_FILTER limit the base filter to the security group. Ex: OU=Superdesk Users,dc=sourcefabric,dc=org
LDAP_BASE_FILTER = env("LDAP_BASE_FILTER", "")

# change the user depending on the LDAP directory structure
LDAP_USER_FILTER = env("LDAP_USER_FILTER", "(&(objectCategory=user)(objectClass=user)(sAMAccountName={}))")

# LDAP User Attributes to fetch. Keys would be LDAP Attribute Name and Value would be Supderdesk Model Attribute Name
LDAP_USER_ATTRIBUTES = json.loads(
    env(
        "LDAP_USER_ATTRIBUTES",
        '{"givenName": "first_name", "sn": "last_name", '
        '"displayName": "display_name", "mail": "email", '
        '"ipPhone": "phone"}',
    )
)

if LDAP_SERVER:
    INSTALLED_APPS.append("apps.ldap")
else:
    INSTALLED_APPS.append("superdesk.users")
    INSTALLED_APPS.append("apps.auth.db")


INSTALLED_APPS.extend(
    [
        "superdesk.upload",
        "superdesk.sequences",
        "superdesk.notification",
        "superdesk.activity",
        "superdesk.vocabularies",
        "superdesk.backend_meta",
        "apps.comments",
        "superdesk.io",
        "superdesk.io.feeding_services",
        "superdesk.io.feed_parsers",
        "superdesk.io.subjectcodes",
        "superdesk.io.iptc",
        "apps.io",
        "apps.io.feeding_services",
        "superdesk.publish",
        "superdesk.commands",
        "superdesk.locators",
        "superdesk.data_updates",
        "apps.auth",
        "apps.archive",
        "apps.stages",
        "apps.desks",
        "apps.tasks",
        "apps.preferences",
        "apps.spikes",
        "apps.prepopulate",
        "apps.legal_archive",
        "apps.search",
        "apps.saved_searches",
        "apps.privilege",
        "apps.rules",
        "apps.highlights",
        "apps.products",
        "apps.publish",
        "apps.publish.enqueue",
        "apps.publish.formatters",
        "apps.content_filters",
        "apps.content_types",
        "apps.dictionaries",
        "apps.duplication",
        "apps.spellcheck",
        "apps.templates",
        "apps.archived",
        "apps.validators",
        "apps.validate",
        "apps.workspace",
        "apps.macros",
        "apps.archive_broadcast",
        "apps.search_providers",
        "apps.vidible",
        "apps.feature_preview",
        "apps.workqueue",
        "apps.languages",
    ]
)

INSTALLED_PLUGINS = ["analytics", "planning"]

INSTALLED_APPS.extend(INSTALLED_PLUGINS)

RENDITIONS = {
    "picture": {
        "thumbnail": {"width": 220, "height": 120},
        "viewImage": {"width": 640, "height": 640},
        "baseImage": {"width": 1400, "height": 1400},
    },
    "avatar": {
        "thumbnail": {"width": 60, "height": 60},
        "viewImage": {"width": 200, "height": 200},
    },
}

SERVER_DOMAIN = "localhost"

MACROS_MODULE = env("MACROS_MODULE", "superdesk.macros")

WS_HOST = env("WSHOST", "0.0.0.0")
WS_PORT = env("WSPORT", "5100")

REDIS_URL = env("REDIS_URL", "redis://localhost:6379")
if env("REDIS_PORT"):
    REDIS_URL = env("REDIS_PORT").replace("tcp:", "redis:")
BROKER_URL = env("CELERY_BROKER_URL", REDIS_URL)

# Determines if the ODBC publishing mechanism will be used, If enabled then pyodbc must be installed along with it's
# dependencies
ODBC_PUBLISH = env("ODBC_PUBLISH", None)
# ODBC test server connection string
ODBC_TEST_CONNECTION_STRING = env(
    "ODBC_TEST_CONNECTION_STRING",
    "DRIVER=FreeTDS;DSN=NEWSDB;UID=???;PWD=???;DATABASE=News",
)

DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES = "AAP"
DEFAULT_URGENCY_VALUE_FOR_MANUAL_ARTICLES = 3

# This value gets injected into NewsML 1.2 and G2 output documents.
NEWSML_PROVIDER_ID = "aap.com.au"
ORGANIZATION_NAME = "Australian Associated Press"
ORGANIZATION_NAME_ABBREVIATION = "AAP"

AMAZON_CONTAINER_NAME = env("AMAZON_CONTAINER_NAME", "")
AMAZON_ACCESS_KEY_ID = env("AMAZON_ACCESS_KEY_ID", "")
AMAZON_SECRET_ACCESS_KEY = env("AMAZON_SECRET_ACCESS_KEY", "")
AMAZON_REGION = env("AMAZON_REGION", "us-east-1")
AMAZON_SERVE_DIRECT_LINKS = env("AMAZON_SERVE_DIRECT_LINKS", False)
AMAZON_S3_USE_HTTPS = env("AMAZON_S3_USE_HTTPS", False)

is_testing = os.environ.get("SUPERDESK_TESTING", "").lower() == "true"
ELASTICSEARCH_FORCE_REFRESH = is_testing
ELASTICSEARCH_AUTO_AGGREGATIONS = True

# URL of the Highcharts Export Server (used in analytics for generating charts on the server)
HIGHCHARTS_SERVER_HOST = env("HIGHCHARTS_SERVER_HOST", "localhost")
HIGHCHARTS_SERVER_PORT = env("HIGHCHARTS_SERVER_PORT", "6060")
HIGHCHARTS_SERVER_WORKERS = env("HIGHCHARTS_SERVER_WORKERS", "1")
HIGHCHARTS_SERVER_WORK_LIMIT = env("HIGHCHARTS_SERVER_WORK_LIMIT", "10")
HIGHCHARTS_SERVER_LOG_LEVEL = env("HIGHCHARTS_SERVER_LOG_LEVEL", "4")
HIGHCHARTS_SERVER_QUEUE_SIZE = env("HIGHCHARTS_SERVER_QUEUE_SIZE", "10")
HIGHCHARTS_SERVER_RATE_LIMIT = env("HIGHCHARTS_SERVER_RATE_LIMIT", False)
ANALYTICS_ENABLE_SCHEDULED_REPORTS = strtobool(env("ANALYTICS_ENABLE_SCHEDULED_REPORTS", "false"))

# Archive Statistics
STATISTICS_MONGO_DBNAME = "sptests"
ANALYTICS_ENABLE_ARCHIVE_STATS = strtobool(env("ANALYTICS_ENABLE_ARCHIVE_STATS", "true"))
